addons/stopwatch/stopwatch.lua

```diff
@@ -0,0 +1,87 @@
+_addon.name = 'stopwatch'
+_addon.author = 'Patrick Finnigan (Puhfyn@Ragnarok)'
+_addon.version = '1.0'
+_addon.commands = {'sw', 'stopwatch'}
+
+require('logger')
+config = require('config')
+texts = require('texts')
+ 
+defaults = {}
+defaults.pos = {}
+defaults.pos.x = 450
+defaults.pos.y = 0
+defaults.text = {}
+defaults.text.font = 'Arial'
+defaults.text.size = 12
+ 
+settings = config.load(defaults)
+times = texts.new(settings)
+startTime = os.time()
+isActive = false
+cumulativeSeconds = 0
+
+
+windower.register_event('prerender', function()
+    local info = windower.ffxi.get_info()
+    if not info.logged_in then
+        times:hide()
+        return
+    end
+
+    seconds = cumulativeSeconds
+    if isActive == true then
+        seconds = seconds + os.time() - startTime
+    end
+    times:text(os.date('!%H:%M:%S', seconds))
```

Comment:
```os.date()``` has some funny corner cases, and you should probably avoid using it here, since it's not really intended to be used for this purpose. The most obvious one in this case is that the displayed time will roll over to 00:00:00 if the stopwatch is ever left running for more than 24 hours, but there are some other weird and subtle things that can happen on Windows. Instead, I suggest you just do the calculation manually, as it's not too difficult.
```lua
times:text(string.format('%0.2d:%0.2d:%0.2d', math.floor(seconds / 3600),
    math.floor(seconds / 60) % 60, math.floor(seconds) % 60))
```
This is also probably a little bit faster, though that's of minimal importance here.